File: mobile/lib/models/incident.dart
/// Mirrors IncidentDto from the ASP.NET Core API.
class Incident {
  const Incident({
    required this.id,
    required this.title,
    required this.description,
    required this.type,
    required this.severity,
    required this.status,
    required this.latitude,
    required this.longitude,
    required this.affectedRadiusMeters,
    required this.reportedAt,
    this.district,
    this.addressText,
    this.estimatedAffectedPeople,
    this.aiSeverity,
    this.aiSeverityScore,
    this.aiConfidence,
    this.aiRationale,
    this.imageCount = 0,
    this.distanceKm,
  });

  final String id;
  final String title;
  final String description;
  final String type;
  final String severity;
  final String status;
  final double latitude;
  final double longitude;
  final int affectedRadiusMeters;
  final DateTime reportedAt;
  final String? district;
  final String? addressText;
  final int? estimatedAffectedPeople;
  final String? aiSeverity;
  final int? aiSeverityScore;
  final double? aiConfidence;
  final String? aiRationale;
  final int imageCount;
  final double? distanceKm;

  bool get isAnalysed => aiSeverityScore != null;

  factory Incident.fromJson(Map<String, dynamic> json) => Incident(
        id: json['id'] as String,
        title: json['title'] as String? ?? 'Untitled incident',
        description: json['description'] as String? ?? '',
        type: json['type'] as String? ?? 'Other',
        severity: json['severity'] as String? ?? 'Low',
        status: json['status'] as String? ?? 'Reported',
        latitude: (json['latitude'] as num).toDouble(),
        longitude: (json['longitude'] as num).toDouble(),
        affectedRadiusMeters: (json['affectedRadiusMeters'] as num?)?.toInt() ?? 1000,
        reportedAt:
            DateTime.tryParse(json['reportedAt'] as String? ?? '')?.toLocal() ??
                DateTime.now(),
        district: json['district'] as String?,
        addressText: json['addressText'] as String?,
        estimatedAffectedPeople: (json['estimatedAffectedPeople'] as num?)?.toInt(),
        aiSeverity: json['aiSeverity'] as String?,
        aiSeverityScore: (json['aiSeverityScore'] as num?)?.toInt(),
        aiConfidence: (json['aiConfidence'] as num?)?.toDouble(),
        aiRationale: json['aiRationale'] as String?,
        imageCount: (json['imageCount'] as num?)?.toInt() ?? 0,
        distanceKm: (json['distanceKm'] as num?)?.toDouble(),
      );
}

File: backend/RescueSriLanka.Api/Features/ComponentA/Models/Incident.cs
using System.ComponentModel.DataAnnotations;

namespace RescueSriLanka.Api.Features.ComponentA.Models;
/// <summary>
/// A reported disaster event. The central entity of Component A — its location
/// and severity feed the citizen map, the safety zones, and every downstream
/// component's resource and team matching.
/// </summary>
public class Incident
{
    public Guid Id { get; set; } = Guid.NewGuid();

    [MaxLength(200)]
    public required string Title { get; set; }

    [MaxLength(4000)]
    public required string Description { get; set; }

    public IncidentType Type { get; set; }

    /// <summary>Severity in force — the coordinator's decision once overridden.</summary>
    public IncidentSeverity Severity { get; set; } = IncidentSeverity.Moderate;

    public IncidentStatus Status { get; set; } = IncidentStatus.Reported;

    // ---------- location ----------
    public double Latitude { get; set; }
    public double Longitude { get; set; }

    /// <summary>Radius of the affected area, used to draw the derived safety zone.</summary>
    public int AffectedRadiusMeters { get; set; } = 1000;

    [MaxLength(100)]
    public string? District { get; set; }

    [MaxLength(300)]
    public string? AddressText { get; set; }

    // ---------- impact ----------
    public int? EstimatedAffectedPeople { get; set; }

    // ---------- Incident Analysis Agent output ----------
    /// <summary>Severity the agent proposed, kept separate from the value in force.</summary>
    public IncidentSeverity? AiSeverity { get; set; }

    /// <summary>Agent score from 0-100. Null until the incident has been analysed.</summary>
    public int? AiSeverityScore { get; set; }

    public double? AiConfidence { get; set; }

    [MaxLength(2000)]
    public string? AiRationale { get; set; }

    public DateTime? AiAnalysedAt { get; set; }

    /// <summary>Set when a coordinator disagreed with the agent.</summary>
    public Guid? SeverityOverriddenBy { get; set; }
    public DateTime? SeverityOverriddenAt { get; set; }

    // ---------- provenance ----------
    public Guid? ReportedByUserId { get; set; }
    public DateTime ReportedAt { get; set; } = DateTime.UtcNow;

    public Guid? VerifiedByUserId { get; set; }
    public DateTime? VerifiedAt { get; set; }

    public DateTime? ResolvedAt { get; set; }

    /// <summary>
    /// When the district-wide warning email went out. Verification and AI
    /// approval both fire the warning, and an incident is often both verified
    /// and approved — this is what stops the district being emailed twice about
    /// the same event.
    /// </summary>
    public DateTime? DistrictWarningSentAt { get; set; }

    /// <summary>False once resolved or rejected — drives the "active" map layer.</summary>
    public bool IsActive { get; set; } = true;

    // ---------- audit ----------
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }

    public ICollection<IncidentImage> Images { get; set; } = [];
}
